// sappquotes/dist/index.js
import fetch from "node-fetch";
export var Categories;
(function (Categories) {
    Categories["inspiration"] = "inspiration";
    Categories["motivation"] = "motivation";
    Categories["love"] = "love";
    Categories["success"] = "success";
    Categories["happy"] = "happy";
    Categories["perseverance"] = "perseverance";
    Categories["wisdom"] = "wisdom";
    Categories["sad"] = "sad";
})(Categories || (Categories = {}));
const url = "https://raw.githubusercontent.com/SapphireDevs/quotes/main/quotes.ts";
export async function fetchCategories() {
    return Object.values(Categories);
}
export async function fetchQuotes() {
    const response = await fetch(url);
    const fileContent = await response.text();
    const regex = /const quotes: Quote\[\] = (\[[\s\S]*?\]);/;
    const match = fileContent.match(regex);
    if (!match || !match[1]) {
        throw new Error("Failed to parse quotes file");
    }
    const quotes = eval(match[1]);
    return quotes;
}
export async function getRandomQuote(category) {
    const quotes = await fetchQuotes();
    let filteredQuotes = quotes;
    if (category) {
        const categories = await fetchCategories();
        const lowercaseCategory = category.toLowerCase();
        if (!categories.map((c) => c.toLowerCase()).includes(lowercaseCategory)) {
            throw new Error(`Invalid category: ${category}`);
        }
        filteredQuotes = quotes.filter((quote) => quote.category.toLowerCase() === lowercaseCategory);
    }
    if (filteredQuotes.length === 0) {
        return {
            author: "Unknown",
            quote: `No quotes found for the specified category: ${category}`,
            category: category,
        };
    }
    const randomIndex = Math.floor(Math.random() * filteredQuotes.length);
    return filteredQuotes[randomIndex];
}
